packages/components/config/playwright/setup/utils/Stickersheet.ts

```diff
@@ -0,0 +1,177 @@
+/* eslint-disable no-await-in-loop */
+/* eslint-disable no-restricted-syntax */
+import { Locator } from '@playwright/test';
+import type { ComponentsPage } from '..';
+
+type AttributesType = Record<string, string>;
+
+class StickerSheet {
+  private componentPage: ComponentsPage;
+
+  private tagname: string;
+
+  private assertion?: (component?: Locator) => Promise<any>;
+
+  private component?: Locator;
+
+  private children?: any;
+
+  private attributes: AttributesType = {};
+
+  private rowId: number = 1;
+
+  constructor(componentPage: ComponentsPage, tagName: string, assertion?: (component?: Locator) => Promise<any>) {
+    this.componentPage = componentPage;
+    this.tagname = tagName;
+    this.assertion = assertion;
+  }
+
+  /**
+   * Sets an assertion function to be executed after a component is added to the sheet.
+   * This will allow to wait for a specific condition to be met before proceeding.
+   * @param assertion - A function that takes an optional Locator and returns a Promise.
+   */
+  public setAssertion(assertion: any) {
+    this.assertion = assertion;
+  }
+
+  /**
+   * Sets the children for the component
+   * @param children - children for the component.
+   */
+  public setChildren(children: any) {
+    this.children = children;
+  }
+
+  /**
+   * Sets the attributes for the component
+   * @param attributes - An object containing attributes for the component.
+   */
+  public setAttributes(attributes: AttributesType) {
+    this.attributes = attributes;
+  }
+
+  /**
+   * Adds a new component to the sheet with specified attributes.
+   */
+  private async addComponentToSheet() {
+    const childrenEl = this.children
+      ? `<${this.tagname} id='${this.tagname}-${this.rowId}'>${this.children}</${this.tagname}>`
+      : `<${this.tagname} id='${this.tagname}-${this.rowId}'></${this.tagname}>`;
```

Comment:
this has to be done in the above mentioned way and not as the following. `children = ''` would still create a value within the slot, and we don't want that for icon button
```suggestion
    const childrenEl = `<${this.tagname} id='${this.tagname}-${this.rowId}'>${this.children || ''}</${this.tagname}>`;
```